torch/csrc/distributed/c10d/reducer.hpp

```diff
@@ -27,11 +27,27 @@ constexpr int kDefaultFirstBucketBytes = int(1024 * 1024);
 constexpr int kDefaultBucketBytesCap = int(25 * 1024 * 1024);
 // Collect runtime stats once for every kDDPRuntimeLoggingSampleRate iterations.
 constexpr int kDDPRuntimeLoggingSampleRate = 100;
+constexpr int kUnsetTime = -1;
+
+inline int64_t current_time_in_nanos() {
+  return torch::autograd::profiler::getTime();
+}
 
 // Forward declaration
 class Logger;
 
 class TORCH_API Timer {
+  private:
+    // The timestamp of forward call start time in each iteration.
+    int64_t forward_start_time = kUnsetTime;
```

Comment:
[Optional] Actually it is a mixed coding style. Personally prefer also using c10::optional<int64_t> instead of int64_t here, so no need create a `kUnsetTime` and have separate processing (that maps `kUnsetTime` to `nullopt`).

I understand this is not new code brought in by this PR. No need to do this code improvement in this PR.

Reply:
Filed https://github.com/pytorch/pytorch/issues/62862, it can be a good ramp up task